File: e092.py
# Problem 92: Square digits chains

from time import time

tic = time()
# add square of digits
def f(n): return sum(int(i)**2 for i in str(n))


count = 0
for start in range(1, 10000000):
    n = start
    while not (n == 1 or n == 89):
        n = f(n)
    if n == 89:
        count += 1
print(f'Arrived at eigthy-nine {count} times.')

print(f'Time elapsed: {time() - tic} seconds')

File: e092.js

let count = 0;
for (let start = 1; start < 10000000; start++) {
    let n = start;
    while (!(n === 1 || n === 89)) n = func(n);
    if (n === 89) count++;
}
console.log(`Arrived at eighty-nine ${count} times.`);

/**Add square of digits */
function func(n) {
    let sum = 0;
    const m = Array.from(n.toString()).map(i => i**2);
    m.forEach(i => sum += i);
    return sum;
}